File: lib/src/numfmt.rs
use crate::Float;
use crate::Int;

pub trait NumberKind: Default + Copy + Clone + PartialEq {}
pub trait NumberFormat<TNum>: Default + Copy + Clone {}

#[derive(Debug, Copy, Clone)]
pub enum Endianness {
    Big,
    Little,
}

impl Default for Endianness {
    fn default() -> Self {
        // contorversial default choice
        Self::Little
    }
}

#[derive(Debug, Copy, Clone)]
pub enum IntFormat {
    Octal,
    Hex,
    Decimal,
    Binary,
}

impl NumberFormat<Int> for IntFormat {}

impl Default for IntFormat {
    fn default() -> Self {
        Self::Binary
    }
}

#[derive(Debug, Copy, Clone)]
pub enum FloatFormat {
    Full,
    Scientific,
    Precision(usize),
}

impl NumberFormat<Float> for FloatFormat {}

impl Default for FloatFormat {
    fn default() -> Self {
        Self::Full
    }
}

File: core/numfmt.py
from enum import Enum
from typing import Literal


class Endianess(Enum):
    LITTLE = 0
    BIG = 1

    def to_literal(self) -> Literal["little"] | Literal["big"]:
        if self == Endianess.LITTLE:
            return "little"
        else:
            return "big"


class IntFmt(Enum):
    DECIMAL = 0
    BINARY = 1
    OCTAL = 2
    HEX = 3


class FloatFmt(Enum):
    FULL = 0
    SCIENTIFIC = 1
